docs/state.md

```diff
@@ -1,3 +1,50 @@
-The library provides multiple helpers to manage UAST tree state.
+The UAST has tree structure which is difficult to operate on and nodes don't have unique identifier.
 
-TODO: document them.
+That's why components require transformation to flat object.
```

Comment:
Suggestion:
```
All components use flat object representation of UAST, instead of default UAST tree structure.

The rationale is that in default UAST tree structure:
 - nodes don't have unique identifiers
 - it is quite complex and had to operate on

For example, original JSON with 6 nodes:
 ...

In flat object representation will look like:
 ...
 
This library provides a build-in [function](#!/JSON%20convertor) to convert UAST to a flat object and multiple helpers to manage UAST flat object state.
```